src/com/facebook/buck/android/MergeAndroidResourcesStep.java

```diff
@@ -216,29 +230,47 @@ public StepExecutionResult execute(ExecutionContext context)
               filesystem,
               useOldStyleableFormat);
 
+      ImmutableSet.Builder<String> requiredPackages = ImmutableSet.<String>builder();
+
+      // Create a temporary list to avoid concurrent modification problems.
+      ArrayList<Entry<String, RDotTxtEntry>> entries =
+          new ArrayList<>(rDotJavaPackageToResources.entries());
+
+      if (skipPrebuiltRDotJava) {
+        // If skip_prebuilt_r_dot_java is true remove all packages except union package
+        Preconditions.checkArgument(
+            unionPackage.isPresent(),
+            "union_package should be specified if skip_prebuilt_r_dot_java is set");
+        rDotJavaPackageToResources = TreeMultimap.create();
+
+      } else {
+        requiredPackages.addAll(symbolsFileToRDotJavaPackage.values());
+      }
+
       // If a resource_union_package was specified, copy all resource into that package,
       // unless they are already present.
       if (unionPackage.isPresent()) {
         String unionPackageName = unionPackage.get();
+        requiredPackages.add(unionPackageName);
+
         // Create a temporary list to avoid concurrent modification problems.
```

Comment:
Looks like this comment is no longer relevant?

Reply:
yes removed the comment.